internal/worker/annotator.go

```diff
@@ -44,14 +47,25 @@ func NewAnnotator(poolSize int, orasClient api.ORASClientInterface, loginURL str
 	}, nil
 }
 
-// AnnotateManifests annotates a list of digests (tags and manifests) concurrently and returns a count of annotated tags & manifests and the first error occurred.
-func (a *Annotator) Annotate(ctx context.Context, digests *[]string) (int, error) {
-	jobs := make([]job, len(*digests))
-	for i, digest := range *digests {
-		jobs[i] = newAnnotateJob(a.loginURL, a.repoName, a.artifactType, a.annotations, a.orasClient, digest)
-	}
+// AnnotateManifests annotates a list of manifests concurrently and returns a count of annotated images and the first error occurred.
+func (a *Annotator) Annotate(ctx context.Context, manifests []string) (int, error) {
+	var annotatedImages atomic.Int64
+	group := a.pool.NewGroup()
 
-	return a.process(ctx, &jobs)
+	for _, digest := range manifests {
+		group.SubmitErr(func() error {
```

Comment:
Because Go closures capture the loop variable by reference, wouldn't every submitted function end up seeing the last manifest value?